lib/compilers/clang.js

```diff
@@ -184,3 +184,17 @@ export class ClangHipCompiler extends ClangCompiler {
         return ['-o', this.filename(outputFilename), '-g1', '--no-gpu-bundle-output', filters.binary ? '-c' : '-S'];
     }
 }
+
+export class ClangIntelCompiler extends ClangCompiler {
+    static get key() {
+        return 'clang-intel';
+    }
+
+    runExecutable(executable, executeParameters, homeDir) {
+        executeParameters.env = {
+            OCL_ICD_FILENAMES: path.dirname(this.compiler.exe) + '/../lib/x64/libintelocl.so',
```

Comment:
can you make this a `path.resolve(path.dirname(this.compiler.exe) + '/../lib/x64/libintelocl.so')` so that it makes a nice path without dots

Reply:
changed. I haven't used js in 10 years and never really learned it.